slam-libraries/viam-cartographer/src/slam_service/config_test.cc

```diff
@@ -0,0 +1,338 @@
+        BOOST_TEST(slamService.translation_weight == parameter_value,
+                   tolerance);
+    } else if (parameter == "rotation_weight") {
+        BOOST_TEST(slamService.rotation_weight == parameter_value, tolerance);
+    }
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteCartoConfigParam_invalid_parameter) {
+    SLAMServiceImpl slamService;
+    std::string parameter = "invalid_parameter";
+    int parameter_value = 999;
+    slamService.config_params =
+        "{mode=2d," + parameter + "=" + std::to_string(parameter_value) + "}";
+    const std::string message =
+        "unknown cartographer config parameter: " + parameter;
+    BOOST_CHECK_EXCEPTION(OverwriteCartoConfigParam(slamService, parameter),
+                          std::runtime_error,
+                          [&message](const std::runtime_error& ex) {
+                              BOOST_CHECK_EQUAL(ex.what(), message);
+                              return true;
+                          });
+}
+
+BOOST_AUTO_TEST_CASE(OverwriteCartoConfigParam_valid_parameters) {
+    SLAMServiceImpl slamService;
+    std::map<std::string, int> param_value_map_ints{
+        {"optimize_every_n_nodes", 1},
+        {"num_range_data", 2},
+        {"max_submaps_to_keep", 33},
+        {"fresh_submaps_count", 4},
+        {"min_added_submaps_count", 5}};
+    std::map<std::string, float> param_value_map_floats{
+        {"missing_data_ray_length", 6.0},
+        {"max_range", 7.0},
+        {"min_range", 8.0}};
+    std::map<std::string, double> param_value_map_doubles{
+        {"min_covered_area", 9.0},
+        {"occupied_space_weight", 10.0},
+        {"rotation_weight", 11.0}};
+    for (auto const& x : param_value_map_ints) {
+        std::string parameter = x.first;
+        int parameter_value = x.second;
+        slamService.config_params = "{mode=2d," + parameter + "=" +
+                                    std::to_string(parameter_value) + "}";
+        OverwriteCartoConfigParam(slamService, parameter);
+        checkCartoConfigParams<int>(slamService, parameter, parameter_value);
+    }
+    for (auto const& x : param_value_map_floats) {
+        std::string parameter = x.first;
+        int parameter_value = x.second;
+        slamService.config_params = "{mode=2d," + parameter + "=" +
+                                    std::to_string(parameter_value) + "}";
+        OverwriteCartoConfigParam(slamService, parameter);
+        checkCartoConfigParams<float>(slamService, parameter, parameter_value);
+    }
+    for (auto const& x : param_value_map_doubles) {
+        std::string parameter = x.first;
+        int parameter_value = x.second;
+        slamService.config_params = "{mode=2d," + parameter + "=" +
+                                    std::to_string(parameter_value) + "}";
+        OverwriteCartoConfigParam(slamService, parameter);
+        checkCartoConfigParams<double>(slamService, parameter, parameter_value);
+    }
+}
```

Comment:
[q] Curious if I can/should do this more elegantly? If so, how?

Reply:
I think this is fine. I might have slightly preferred not using the for loops, since it doesn't save much code and makes things slightly less explicit.

Reply:
I think you wouldn't need checkCartoConfigParams() without the for loops. But definitely not worth doing the work :)